test_conformance/subgroups/test_workgroup.cpp

```diff
@@ -621,234 +621,85 @@ template <typename Ty> struct BC
+                                          num_elements_, "test_bcast",
+                                          bcast_source, 0, useCoreSubgroups_);
+        error |= test<T, RED<T, 0>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_redadd",
+            redadd_source, 0, useCoreSubgroups_);
+        error |= test<T, RED<T, 1>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_redmax",
+            redmax_source, 0, useCoreSubgroups_);
+        error |= test<T, RED<T, 2>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_redmin",
+            redmin_source, 0, useCoreSubgroups_);
+        error |= test<T, SCIN<T, 0>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scinadd",
+            scinadd_source, 0, useCoreSubgroups_);
+        error |= test<T, SCIN<T, 1>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scinmax",
+            scinmax_source, 0, useCoreSubgroups_);
+        error |= test<T, SCIN<T, 2>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scinmin",
+            scinmin_source, 0, useCoreSubgroups_);
+        error |= test<T, SCEX<T, 0>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scexadd",
+            scexadd_source, 0, useCoreSubgroups_);
+        error |= test<T, SCEX<T, 1>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scexmax",
+            scexmax_source, 0, useCoreSubgroups_);
+        error |= test<T, SCEX<T, 2>, G, L>::run(
+            device_, context_, queue_, num_elements_, "test_scexmin",
+            scexmin_source, 0, useCoreSubgroups_);
+        return error;
+    }
+
+    cl_device_id device_;
+    cl_context context_;
+    cl_command_queue queue_;
+    int num_elements_;
+    bool useCoreSubgroups_;
+};
 
 // Entry point from main
 int test_work_group_functions(cl_device_id device, cl_context context,
                               cl_command_queue queue, int num_elements,
                               bool useCoreSubgroups)
 {
     int error;
-
-    // Adjust these individually below if desired/needed
-#define G 2000
-#define L 200
-
     error = test<int, AA<0>, G, L>::run(device, context, queue, num_elements,
                                         "test_any", any_source, 0,
                                         useCoreSubgroups);
     error |= test<int, AA<1>, G, L>::run(device, context, queue, num_elements,
                                          "test_all", all_source, 0,
                                          useCoreSubgroups);
+    run_for_type rft(device, context, queue, num_elements, useCoreSubgroups);
+    error |= rft.run<cl_uint>();
+    error |= rft.run<cl_int>();
+    error |= rft.run<cl_ulong>();
+    error |= rft.run<cl_long>();
+    error |= rft.run<float>();
+    error |= rft.run<double>();
+    // error |= rft.run<cl_half>();
```

Comment:
We should create an issue to look into this. Why was half disabled?

Reply:
Regarding half type - I do not know why this is commented (I just leave this as is for today) However I have implemented/fixed this in code related with subgroups next.

Reply:
I think it's worth creating an issue to make sure we don't forget about this.

Reply:
Done #812